src/seml/experiment/mattermost_observer.py

```diff
@@ -120,36 +124,43 @@ def __init__(
             'Next heartbeat will be sent in about _{heartbeat_interval}_, i.e., on _{next_heartbeat_date}_.'
         )
 
-        self.run = None
+        self._run: dict[str, Any] | None = None
         self.channel = channel
 
         self.notify_on_completed = notify_on_completed
         self.notify_on_failed = notify_on_failed
         self.notify_on_interrupted = notify_on_interrupted
         self.notify_on_started = notify_on_started
         self.notify_on_heartbeat = False
-        self.last_heartbeat_notification = None
+        self.last_heartbeat_notification = utcnow()
         self.convert_utc_to_local_timezone = convert_utc_to_local_timezone
 
         self.heartbeat_interval = None
         if heartbeat_interval is not None:
             # unfortunately datetime.strptime() doesn't work with timedeltas, so we parse the date ourselves:
             pattern = re.compile('([0-9]+)-([0-9]+):([0-9]+)')
-            days, hours, minutes = pattern.match(heartbeat_interval).groups()
+            match = pattern.match(heartbeat_interval)
+            assert match is not None, 'Invalid heartbeat_interval format. Use D-hh:mm.'
+            days, hours, minutes = match.groups()
             self.heartbeat_interval = timedelta(
                 days=int(days), hours=int(hours), minutes=int(minutes)
             )
             self.notify_on_heartbeat = True
 
+    @property
```

Comment:
ok, what's the rationale behind making the run get only?

Reply:
compatibility with the super class